incubator/distribution/templates/hooks/create-user.yaml

```diff
@@ -0,0 +1,43 @@
+{{- if .Values.mongodb.enabled }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ .Release.Name }}-create-user"
+  labels:
+    app: {{ template "distribution.fullname" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    component: mongodb
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:
+      labels:
+        app: {{ template "distribution.name" . }}
+        chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+        heritage: {{ .Release.Service }}
+        release: {{ .Release.Name }}
+    spec:
+      {{- if .Values.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.nodeSelector | indent 8 }}
+      {{- end }}
+      restartPolicy: OnFailure
+      containers:
+      - name: post-install-job
+        image: "mvertes/alpine-mongo:{{ .Values.mongodb.image.tag }}-0"
+        command:
+        - 'sh'
+        - '-c'
+        - 'sh /scripts/setup.sh'
```

Comment:
Do you have issues with this running? Configmaps are mounted readonly.

Reply:
@rjkernick  It works for me on k8s 1.9.6.